moveit_ros/planning/plan_execution/src/plan_execution.cpp

```diff
@@ -401,7 +401,7 @@ moveit_msgs::msg::MoveItErrorCodes plan_execution::PlanExecution::executeAndMoni
     plan.plan_components[component_idx].trajectory->getRobotTrajectoryMsg(msg);
     if (!trajectory_execution_manager_->push(msg, plan.plan_components[component_idx].controller_name))
     {
-      trajectory_execution_manager_->clear();
+      trajectory_execution_manager_->stopExecution(true);
```

Comment:
Here as well, neither clear() nor stopExecution should be called. If push failed (because another execution is still ongoing), we should not cancel this other execution.
```suggestion
```